features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/net/BatchesToSegmentsMapper.kt

```diff
@@ -114,31 +131,38 @@ internal class BatchesToSegmentsMapper {
             source = MobileSegment.Source.ANDROID,
             records = emptyList()
         )
-        val segmentAsJsonObject = segment.toJson().safeGetAsJsonObject() ?: return null
+        val segmentAsJsonObject = segment.toJson().safeGetAsJsonObject(internalLogger)
+            ?: return null
         segmentAsJsonObject.add(RECORDS_KEY, orderedRecords)
         return Pair(segment, segmentAsJsonObject)
     }
 
     private fun hasFullSnapshotRecord(records: JsonArray) =
         records.firstOrNull {
-            it.asJsonObject.getAsJsonPrimitive(RECORD_TYPE_KEY)?.safeGetAsLong() ==
+            it.asJsonObject.getAsJsonPrimitive(RECORD_TYPE_KEY)?.safeGetAsLong(internalLogger) ==
                 FULL_SNAPSHOT_RECORD_TYPE
         } != null
 
     private fun JsonObject.records(): JsonArray? {
-        return get(EnrichedRecord.RECORDS_KEY)?.safeGetAsJsonArray()
+        return get(EnrichedRecord.RECORDS_KEY)?.safeGetAsJsonArray(internalLogger)
     }
 
     private fun JsonObject.timestamp(): Long? {
-        return getAsJsonPrimitive(TIMESTAMP_KEY)?.safeGetAsLong()
+        return getAsJsonPrimitive(TIMESTAMP_KEY)?.safeGetAsLong(internalLogger)
     }
 
     private fun JsonObject.rumContext(): SessionReplayRumContext? {
         val applicationId = get(EnrichedRecord.APPLICATION_ID_KEY)?.asString
         val sessionId = get(EnrichedRecord.SESSION_ID_KEY)?.asString
         val viewId = get(EnrichedRecord.VIEW_ID_KEY)?.asString
         if (applicationId == null || sessionId == null || viewId == null) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
+            internalLogger.log(
```

Comment:
Is this telemetry log actionable? Specifically, do we already know what we could do/check if we see those events appearing in the telemetry org? Do we have enough info to deal with this?

Reply:
well, means that we have a problem in the way we asynchronously transmit the RUM context and wanted to make sure we are safe there. Without this we are never going to be sure.